src/data/mystery-encounters/mystery-encounter.ts

```diff
@@ -0,0 +1,963 @@
+   */
+  encounterAnimations?: EncounterAnim[];
+  /**
+   * If specified, defines any game modes where the MysteryEncounter should *NOT* spawn
+   */
+  disabledGameModes?: GameModes[];
+  /**
+   * If true, hides "A Wild X Appeared" etc. messages
+   * Default true
+   */
+  hideBattleIntroMessage: boolean;
+  /**
+   * If true, when an option is selected the field visuals will fade out automatically
+   * Default false
+   */
+  autoHideIntroVisuals: boolean;
+  /**
+   * Intro visuals on the field will slide in from the right instead of the left
+   * Default false
+   */
+  enterIntroVisualsFromRight: boolean;
+  /**
+   * If true, allows catching a wild pokemon during the encounter
+   * Default false
+   */
+  catchAllowed: boolean;
+  /**
+   * If true, encounter will continuously run through multiple battles/puzzles/etc. instead of going to next wave
+   * MUST EVENTUALLY BE DISABLED TO CONTINUE TO NEXT WAVE
+   * Default false
+   */
+  continuousEncounter: boolean;
+  /**
+   * Maximum number of times the encounter can be seen per run
+   * Rogue tier encounters default to 1, others default to 3
+   */
+  maxAllowedEncounters: number;
+  /**
+   * If true, encounter will not animate the target Pokemon as part of battle animations
+   * Used for encounters where it is not a "real" battle, but still uses battle animations and commands (see {@linkcode FunAndGamesEncounter} for an example)
+   */
+  hasBattleAnimationsWithoutTargets: boolean;
+  /**
+   * If true, will skip enemy pokemon turns during battle for the encounter
+   * Used for encounters where it is not a "real" battle, but still uses battle animations and commands (see {@linkcode FunAndGamesEncounter} for an example)
+   */
+  skipEnemyBattleTurns: boolean;
+  /**
+   * If true, will skip COMMAND input and go straight to FIGHT (move select) input menu
+   */
+  skipToFightInput: boolean;
+
+  /**
+   * Event callback functions
+   */
+  /** Event when Encounter is first loaded, use it for data conditioning */
+  onInit?: (scene: BattleScene) => boolean;
+  /** Event when battlefield visuals have finished sliding in and the encounter dialogue begins */
+  onVisualsStart?: (scene: BattleScene) => boolean;
+  /** Event triggered prior to {@linkcode CommandPhase}, during {@linkcode TurnInitPhase} */
+  onTurnStart?: (scene: BattleScene) => boolean;
+  /** Event prior to any rewards logic in {@linkcode MysteryEncounterRewardsPhase} */
+  onRewards?: (scene: BattleScene) => Promise<void>;
+  /** Will provide the player party EXP before rewards are displayed for that wave */
+  doEncounterExp?: (scene: BattleScene) => boolean;
+  /** Will provide the player a rewards shop for that wave */
+  doEncounterRewards?: (scene: BattleScene) => boolean;
+  /** Will execute callback during VictoryPhase of a continuousEncounter */
+  doContinueEncounter?: (scene: BattleScene) => Promise<void>;
+
+  /**
+   * Requirements
+   */
+  requirements: EncounterSceneRequirement[];
```

Comment:
you could use 
```suggestion
  // #region Requirements
  
```

Make sure to close it with
```ts
  // #endregion
```

This would show up like this on vs-codium minimap:
<img width="177" alt="image" src="https://github.com/user-attachments/assets/2d2d9736-b043-49d7-a1b0-a0002e36509a">